SDL fails to build with EXTENDED_MEDIA_MODE=ON

### Bug Report
Currently, SDL fails to build successfully if the `EXTENDED_MEDIA_MODE` flag is set to ON 

##### Reproduction Steps
1. In the build directory, run `cmake ../sdl_core -DEXTENDED_MEDIA_MODE=ON`
2. In the build directory, run `make`

##### Expected Behavior
SDL builds without errors

##### Observed Behavior
SDL build fails with the following error

```
[ 31%] Building CXX object src/components/media_manager/CMakeFiles/MediaManager.dir/src/file_streamer_adapter.cc.o
/home/user/sdl_core/src/components/media_manager/src/audio/from_mic_to_file_recorder_thread.cc: In member function ‘virtual void media_manager::FromMicToFileRecorderThread::exitThreadMain()’:
/home/user/sdl_core/src/components/media_manager/src/audio/from_mic_to_file_recorder_thread.cc:369:19: error: ‘class threads::Thread’ has no member named ‘stop’; did you mean ‘Stop’?
     sleepThread_->stop();
                   ^~~~
                   Stop
src/components/media_manager/CMakeFiles/MediaManager.dir/build.make:127: recipe for target 'src/components/media_manager/CMakeFiles/MediaManager.dir/src/audio/from_mic_to_file_recorder_thread.cc.o' failed
make[2]: *** [src/components/media_manager/CMakeFiles/MediaManager.dir/src/audio/from_mic_to_file_recorder_thread.cc.o] Error 1
make[2]: *** Waiting for unfinished jobs....

```

##### OS & Version Information
* OS/Version: Ubuntu 18.04
* SDL Core Version: develop (https://github.com/smartdevicelink/sdl_core/commit/9ac9acdec6d82b280fca5286549ea904e7bc8b4c)

--- a/src/components/media_manager/src/audio/from_mic_to_file_recorder_thread.cc
+++ b/src/components/media_manager/src/audio/from_mic_to_file_recorder_thread.cc
@@ -366,7 +366,7 @@ void FromMicToFileRecorderThread::exitThreadMain() {
 
   if (sleepThread_) {
     SDL_LOG_DEBUG("Stop sleep thread\n");
-    sleepThread_->stop();
+    sleepThread_->Stop(threads::Thread::kThreadStopDelegate);
   }
 
   SDL_LOG_TRACE("Set should be stopped flag\n");

